### ISSUE TITLE
Proc to return a row of OUT values

### ISSUE THREAD

**Issue by MendesKevin** [vitaly-t/pg-promise#692]
In postgresql [documentation](https://www.postgresql.org/docs/12/sql-createprocedure.html):
> The mode of an argument: IN, INOUT, or VARIADIC. If omitted, the default is IN. (OUT arguments are currently not supported for procedures. Use INOUT instead.)

Procedures are considered like functions who returns nothing but postgresql let you returns something.

And pg-promise use by default querymask 'none' when using proc.

Why ?

**Comment by vitaly-t** [comment]
I'm not sure if output variables are at all supported by the driver. If they are, then please provide an example, using a generic query.

**Comment by MendesKevin** [comment]
_Sorry, i'm not good english speaker and don't understand well what you are asking._

Here is an example:
```
CREATE OR REPLACE PROCEDURE test(INOUT output_value boolean)
LANGUAGE plpgsql AS $$
BEGIN
    output_value := true;
END;$$;
```

It's INOUT argument we must give a value this is why the _null_ value in pg-promise functions parameters.

_for preventing that we can set a default value in procedure definition_
`CREATE OR REPLACE PROCEDURE test(INOUT output_value boolean DEFAULT null)`

`db.proc("test", [ null ])` throw an error because query return a row.
```
QueryResultError {
    code: queryResultErrorCode.notEmpty
    message: "No return data was expected."
    received: 1
    query: "call test(null)"
}
```
Following the documentation, it's normal behavior, proc use
`none('CALL $1:alias($2:csv)', [procName, values])`

For getting procedure working with INOUT arguments we need change querymask to accept at least one row retrieved
`db.oneOrNone('CALL $1:alias($2:csv)', ["test", [ null ]])` return `{ output_value: true }`

pg-promise seems to handle INOUT arguments (OUT currently not supported by postgresql).

_sorry again for my english 🥖_

**Comment by vitaly-t** [comment]
Ok, if the procedure can return one row like that, then the API will need to be revised.

**Comment by vitaly-t** [comment]
Updated functionality with [this commit](https://github.com/vitaly-t/pg-promise/commit/4180abaa0d6d9c160b52be614718e1b43d380d40).

**Comment by vitaly-t** [comment]
The update has been released in [v10.3.3](https://github.com/vitaly-t/pg-promise/releases/tag/10.3.3).
